include/RAJA/policy/cuda/forall.hpp

```diff
@@ -47,76 +47,133 @@
 
 namespace RAJA
 {
-
 namespace policy
 {
-
 namespace cuda
 {
 
 namespace impl
 {
 
-/*!
- ******************************************************************************
- *
- * \brief calculate gridDim from length of iteration and blockDim
- *
- ******************************************************************************
- */
-RAJA_INLINE
-cuda_dim_t getGridDim(cuda_dim_member_t len, cuda_dim_t blockDim)
+template<typename IterationMapping, typename IterationGetter, typename UniqueMarker>
```

Comment:
Should the template parameters have come minimal documentation to explain what they mean. For example, it's not clear to me what UniqueMarker is.

Reply:
That's probably a good idea, I'll try to improve on the documentation that was there before.

Reply:
Added documentation.